Minefields at bridges

Night-Pryanik opened the pull request:
#### Summary
```SUMMARY: Content "Minefields will spawn only at the entry of the bridges. No more minefields spawning in open field or deep in the forest."```

#### Purpose of change
Make minefields great again.

It seems that many people disliked my variant of minefields with random placement of signs. I still think it's better than the old variant with dirt mounds revealing exact locations of mines, but it's still not very good overall. So I decided to restrict minefields spawn only to roads (bridges, more exactly).

#### Describe the solution
- Minefields now spawn only at entry of the bridges - only one variant with bridge going north and road going south for now.
- It's only two warning signs at the lower edge of the submap instead of four signs at every border.
- Added two sections of blocks made of sandbags - one with a chance to spawn humvee, another with a chance to spawn killed soldier. Both sections have 7.62x51 mm casings scattered all over the place simulating soldiers' and/or humvee gunfire.
- Added a chance to spawn dead people and/or zombies on the minefield, either killed by soldiers' gunfire or after stepping on a mine.
- Added a line of barbed wire fence with a chance to spawn another set of people/zombies who managed to cross the minefield but got stuck at the fence and got eventually killed too.
- Also I made generic corpses spawn with items from generic zombie loot list (previously generic corpses spawned without underwear and without any items other than clothing).

#### Describe alternatives you've considered
Don't replace old variant of the minefields with my own, but rather add my variant to the old one.

#### Additional context
![изображение](https://user-images.githubusercontent.com/11132525/58766495-736fc380-8590-11e9-9f06-76b000e871ce.png)

As I said, it's only one variant of bridge entry implemented. I want to know if the direction I'm moving is right and whether I should add another variants of bridge entries (bridge going east, west and south).

anothersimulacrum commented on src/map_extras.cpp:
```diff
@@ -798,32 +798,103 @@ static void mx_portal( map &m, const tripoint &abs_sub )
+
+        for( const auto &loc : g->m.points_in_radius( { 6, 4, abs_sub.z }, 3, 0 ) ) {
+            if( one_in( 4 ) ) {
+                m.add_item_or_charges( loc, item( "762_51_casing" ) );
```
The US military primarily uses 5.56 rounds, so perhaps this should be changed to casings of that variety?

Night-Pryanik replied:
These are casings from humvee's m60.

Night-Pryanik replied:
To the right there is another sandbag block with a dead soldier, so I replaced 7.62 with 5.56 there.

FrozenSnowFox commented:
Now this is a nice improvement a logical highly visible location.  I trust its not at every bridge though because that might be a bit excessive.

You could have similar setups in or near the city.  Safe zone areas with chainlink fencing guarded by soldiers to protect civilians.  Barricade setups outside highly durable, defensive or well known buildings (banks, police stations ect).  FEMA camps would probably be good places for barricade setups.  Some labs above and below ground would likely want to protect themselves better with a military barricade.

Night-Pryanik commented:
> I trust its not at every bridge though because that might be a bit excessive.

No, of course not. It should be quite rare in fact.

> You could have similar setups in or near the city. Safe zone areas with chainlink fencing guarded by soldiers to protect civilians. Barricade setups outside highly durable, defensive or well known buildings (banks, police stations ect). FEMA camps would probably be good places for barricade setups. Some labs above and below ground would likely want to protect themselves better with a military barricade.

Good ideas, though they were already suggested several times. Definitely not in this PR and probably not me.

eilaattwood commented:
@Night-Pryanik One of my characters actually died recently because of your 'improved' minefields, because signs are currently quite hard to notice, so I like this approach much more. :D

Night-Pryanik commented:
Added another variant (bridge going south, road going north).

![изображение](https://user-images.githubusercontent.com/11132525/58798809-ab204f00-8614-11e9-9cf0-40d7363251c5.png)

curstwist commented:
probably a move in the right direction, even if I'll miss the old random placements which could be a boon to early characters survival chances (or early deaths).  Maybe there are some other places we can consider adding them (in future PRs) that are also more appropriate.  I think stuff like the bunkers and outposts should have minefield perimeters, etc.

Night-Pryanik commented:
Added another variant (bridge going west, road going east).

It has "ordinary" sub-variant with intact tents, sandbag and barbed wire barricades. 
![изображение](https://user-images.githubusercontent.com/11132525/58823861-bf337300-864b-11e9-8868-5e677602855d.png)

Also I added another sub-variant with some sick psycho drove right through the tent trying to break through the roadblock.
![изображение](https://user-images.githubusercontent.com/11132525/58823896-d6726080-864b-11e9-9f15-1f3db6e8519b.png)

Night-Pryanik commented:
Added the last variant (bridge going east, road going west).

![изображение](https://user-images.githubusercontent.com/11132525/58869966-4c201000-86d0-11e9-86df-354186824f7e.png)

Now it's time to clean all the mess.

Night-Pryanik commented:
Ready.

Night-Pryanik commented:
Jenkins rebuild
